demos/GMW_demo/gmw_and.js

```diff
@@ -0,0 +1,174 @@
+
+/**
+ * AND on two shares, return secret share of ci.
+ * @function gmw_and
+ * @param {module:jiff-client~JIFFClient} jiff - the jiff instance
+ * @param {number} secret - the secret1 to share.
+ * @param {number} secret - the secret2 to share.
+ * @returns {object} a secret share of value ci
+ */
+
+function gmw_and(jiff,share1,share2) {
+  if (!(share1.jiff === share2.jiff)) {
+    throw new Error('shares do not belong to the same instance (^)');
+  }
+  if (!share1.jiff.helpers.Zp_equals(share1, share2)) {
```

Comment:
I think we should check that both shares have the same field parameter, and specifically that the field is 2, to prevent the error we saw earlier this week when calling with a share from the standard jiff_instance.share() function.